### PR TITLE
PowerBI - Schema Refactoring

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes :

Added fix for PowerBI schema

#
### Type of change :

- [x] Bug fix
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach @pmbrull

### CODE REVIEW

**Comment by ayush-shah** [comment]
I will be rebasing and modifying based on #3959

File: ingestion/src/metadata/ingestion/source/powerbi.py [hunk 1]
```diff
@@ -100,7 +89,12 @@ def create(cls, config_dict, metadata_config: OpenMetadataServerConfig):
         Returns:
             PowerBiSource
         """
-        config = PowerbiSourceConfig.parse_obj(config_dict)
+        config = WorkflowSource.parse_obj(config_dict)
+        connection: PowerBIConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, PowerBIConnection):
+            raise InvalidSourceException(
+                f"Expected RedshiftConnection, but got {connection}"
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/powerbi.py:96]: Use PowerBIConnection here pls 🙏 . It's hard to not miss this 😂  might need to find a better way to name this by passing the class or smt

File: ingestion/src/metadata/ingestion/source/powerbi.py [hunk 2]
```diff
@@ -61,33 +53,30 @@ class PowerbiSource(Source[Entity]):
...
-    status: SourceStatus
-
     def __init__(
         self,
-        config: PowerbiSourceConfig,
+        config: WorkflowSource,
         metadata_config: OpenMetadataServerConfig,
     ):
         super().__init__()
+        print(config)
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/powerbi.py:62]: print!

**Review by pmbrull — APPROVED** [review]
Thanks, a couple of minor comments. LGTM. Let's merge after updating the filterPattern stuff

**Comment by ayush-shah** [comment]
Yes - give me few mins - will push the changes alongwith rebase

File: ingestion/src/metadata/ingestion/source/powerbi.py (hunk 1 above)
Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/source/powerbi.py]: Resolved

File: ingestion/src/metadata/ingestion/source/powerbi.py (hunk 2 above)
Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/source/powerbi.py]: resolved

**Pull request merged**